### DISCUSSION TITLE
List my own permissions

### DISCUSSION THREAD

**Discussion by GrampaWildWilly** [community/community#131653] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I am looking for a place where I can see the list of permissions that have been defined for my own user ID in a given repository.  I have found lots of documentation on what it means to have read permission, triage permission, administrator permission, owner permission.  But I haven't found anywhere that explains how I can see what permissions I have.  Is there such a feature?

**Comment by gaomingzhao666** [comment]
Hey bro.

**You can enter the `settings` panel, click developer settings at the bottom of the buttons, and then select `Personal Access Token`.**

**There you can see the `repository permissions` like the following images.**

![1](https://github.com/community/community/assets/155299070/76189375-143f-446a-a2bf-a119070aa20d)
![2](https://github.com/community/community/assets/155299070/aff3b723-178f-4ea8-af72-3fd10b94abcf)

If my comment helps you solve this question, please mark my comment as an answer, thanks bro. 😊

**Reply by GrampaWildWilly** [reply]
Hmmm.......  Interesting.  I didn't know what that was for.  Plus not being an actual developer of the code in question, I never thought to click the developer settings.  But I generated a token & granted myself read/write permission on every attribute listed.  Then I looked at the token & it showed me what I was allowed to do.  I think if I selected an attribute value that I was not permitted to have, it just ignored my request.  That is, if I clicked read & write, but I wasn't actually permitted to perform that function, it just didn't honor my request.  Do I have that right?  Or does it show read & write if I have that permission but only read-only if I don't?  In that case, I wouldn't have to actually proceed to generating the token.

If this is what you have to do to get something as simple as a list of all my permissions, this is quite complicated, much more than I would have expected.  I was looking for a button somewhere in my profile or my settings that said simply, "Your permissions."  I would click that & I would get a list.  The list would say, "You have triage permission in this repository."  Or whatever might be applicable.  This token generation process is awfully weird, non-intuitive in the extreme.

**Reply by TheGiraffe3** [reply]
I'm not sure that is exactly how it works, but I'm not sure how to check.
Is the repository in a user account or an organization?
If an user's repository, then you can't have different levels of access. All you can have is write.

**Reply by GrampaWildWilly** [reply]
It seems I'm not alone in this boat.  I should think listing your own permissions would be an easy thing to do.

The repository is owned by a user account . . . I think.  I'm not the owner.  The number of levels of access is not what I'm interested in.  I want to know simply what my own permissions are.  I want GitHub to just tell what I've been granted.  I assume these permissions could be & most likely are different from repository to repository.

**Reply by TheGiraffe3** [reply]
If it is a user repository, then you can only write, and do nothing else.
There is probably a way to do it, but GitHub can be obscure sometimes.

**Reply by GrampaWildWilly** [reply]
"can be obscure sometimes."  You are a master of understatement.

The repository does have a code section but I'm mostly interested in the discussions section.  I'd like to see a list of the operations I can perform.  Like mark a post as an answer in a discussion I didn't start, like delete a post I didn't write, like block a user, like lock a discussion, that sort of thing.  I'm looking for that list of permissions.  Failing that, I'm looking for a place that says, you are a simple participant, or you have triage permission, or you have administrator permission.  I have found in the GitHub Docs very good documentation, a nice table with xes & checkmarks, showing what you can & can't do if you're one or the other.  But I can't find a place that says, "You are just a participant, or you are a triager, or you are an administrator."

**Reply by TheGiraffe3** [reply]
Just checked. _If_ you have write access to a user's repository, you can do what you asked in the previous post.

**Reply by gaomingzhao666** [reply, marked as answer]
Hey bro, thank you for your reply.

**If you want to find a place to check your permission to a given repo, it seems can not, you can only do is fork this repo and then you can push some code on this repo by create a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests), but those code must be verified by the repository owner and then can be saved in this repo.**

**I shown the images above is to the owner of a repo, and the owners can check the permission to this repo, and access or act to this repo by using `Personal Access Token`.**

**In conclusion and in my opinion, you have no any permission to a repo that you are not owning, so naturally there is no place to check the permission to that repo. If you are the owner of a repo, you have all of the permission to it, so there is no place to check the permission too.**

**Reply by GrampaWildWilly** [reply]
Thanks @gaomingzhao666 Nano.  That explains what I've been seeing.  It didn't make any sense to me.  I can't say it does now, either, but at least what you say seems to match what I've been seeing.  Thanks for the help.
